src/buffer.rs

```diff
@@ -20,7 +20,10 @@ pub trait Pixel: Copy + Clone {
     type Subpixel: Primitive;
 
     /// Returns the number of channels of this pixel type.
+    #[deprecated(note="please use CHANNEL_COUNT associated constant")]
     fn channel_count() -> u8;
```

Comment:
This should probably be a default impl like:
```rust
fn channel_count() -> u8 {
    return Self::CHANNEL_COUNT;
}
```

Reply:
Should i do the same for the other two constants?